File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left=0;
        int right=numbers.size()-1;
        vector <int> result={-1,-1};

        while (left<right){
            if ((numbers[left]+numbers[right])<target){
                left++ ;
            }
            else if ((numbers[left]+numbers[right])>target){
                right--;
            }
            else {
                result[0] = left+1;
                result[1] = right+1;
                return result;
            }
        }
        
    }
};

File: Data Structures & Algorithms/special-array-i/submission-1.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        if len(nums)<=1:
            return True
        else:
            left=0
            right=1
            flag=False

            while right<len(nums):
                if (nums[left]%2==0 and nums[right]%2!=0) or (nums[left]%2!=0 and nums[right]%2==0):
                    flag=True
                else:
                    return False
                
                left+=1
                right+=1
            
            return flag